engine/execution/pruner/core.go

```diff
@@ -61,15 +61,24 @@ func LoopPruneExecutionDataFromRootToLatestSealed(
 			return fmt.Errorf("failed to get next and latest to prune: %w", err)
 		}
 
+		commitDuration := 2 * time.Millisecond // with default batch size 1200, the avg commit duration is 2ms
+		batchCount, totalDuration := EstimateBatchProcessing(
+			nextToPrune, latestToPrune,
+			config.BatchSize, config.SleepAfterEachBatchCommit, commitDuration)
+
 		log.Info().
 			Uint64("nextToPrune", nextToPrune).
 			Uint64("latestToPrune", latestToPrune).
-			Msgf("execution data pruning will start in %s at %s",
-				config.SleepAfterEachIteration, time.Now().Add(config.SleepAfterEachIteration).UTC())
+			Uint64("batchCount", batchCount).
+			Uint64("totalDuration", uint64(totalDuration.Seconds())).
+			Msgf("execution data pruning will start in %s at %s, complete at %s",
```

Comment:
Add:
- total number of batches
- estimated total duration
- estimated completion time